daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -133,6 +133,11 @@ object Speedy {
       case _ =>
     }
 
+    def setExpressionToEvaluate(expr: SExpr): Unit = {
+      ctrl = expr
+      returnValue = null
```

Comment:
Shouldn't we also reset `kontStack` here? This would be particularly important when you reuse a machine that had failed with an exception.

Reply:
Yes.
And in fact I do in my next PR.

Reply:
If that's an actual problem then let's please fix it in this PR. I'm not keen to have potentially broken things in master if we can avoid it. You never know who runs a `daml-sdk-head` when.